sql/opt/exec/execbuilder: TestExecBuild failed

sql/opt/exec/execbuilder.TestExecBuild [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3738421&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3738421&tab=artifacts#/) on master @ [6e288820935624d0839406a0a8bbacaff8fe960d](https://github.com/cockroachdb/cockroach/commits/6e288820935624d0839406a0a8bbacaff8fe960d):


```
=== RUN   TestExecBuild
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestExecBuild770660939
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestExecBuild
    logic.go:3514: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestExecBuild770660939
--- FAIL: TestExecBuild (99.51s)
=== RUN   TestExecBuild/local/show_trace_nonmetamorphic/system_table_lookup
    logic.go:2441: let $rangeid = 44
--- done: testdata/schema_change_in_txn_nonmetamorphic with config local: 9 tests, 0 failures
--- total progress: 400 statements/queries
=== CONT  TestExecBuild/local/show_trace_nonmetamorphic/system_table_lookup
    logic.go:2400: 
         
        testdata/show_trace_nonmetamorphic:366: SELECT operation, message FROM [SHOW KV TRACE FOR SESSION]
        WHERE message     LIKE '%r44: sending batch%'
          AND message NOT LIKE '%PushTxn%'
          AND message NOT LIKE '%QueryTxn%'
        expected:
            dist sender send  r44: sending batch 6 CPut to (n1,s1):1
            dist sender send  r44: sending batch 6 CPut to (n1,s1):1
            dist sender send  r44: sending batch 6 CPut to (n1,s1):1
            dist sender send  r44: sending batch 6 CPut to (n1,s1):1
            dist sender send  r44: sending batch 1 EndTxn to (n1,s1):1
        but found (query options: "") :
            dist sender send  r44: sending batch 24 CPut, 1 EndTxn to (n1,s1):1
--- done: testdata/show_trace_nonmetamorphic with config local: 48 tests, 1 failures
            --- FAIL: TestExecBuild/local/show_trace_nonmetamorphic/system_table_lookup (4.05s)
=== RUN   TestExecBuild/local/show_trace_nonmetamorphic
=== PAUSE TestExecBuild/local/show_trace_nonmetamorphic
=== CONT  TestExecBuild/local/show_trace_nonmetamorphic
        --- FAIL: TestExecBuild/local/show_trace_nonmetamorphic (12.99s)
=== RUN   TestExecBuild/local
    --- FAIL: TestExecBuild/local (0.02s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/opt/exec/execbuilder.TestExecBuild [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3746353&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3746353&tab=artifacts#/) on master @ [52a7a6f10a3ed02d7d4d0b7dc09406c86ba937dd](https://github.com/cockroachdb/cockroach/commits/52a7a6f10a3ed02d7d4d0b7dc09406c86ba937dd):


```
=== RUN   TestExecBuild
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestExecBuild1715850967
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestExecBuild
    logic.go:3514: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestExecBuild1715850967
--- FAIL: TestExecBuild (92.80s)
=== RUN   TestExecBuild/local/show_trace_nonmetamorphic/system_table_lookup
    logic.go:2441: let $rangeid = 44
--- done: testdata/update_from with config local: 11 tests, 0 failures
--- total progress: 175 statements/queries
=== CONT  TestExecBuild/local/show_trace_nonmetamorphic/system_table_lookup
    logic.go:2400: 
         
        testdata/show_trace_nonmetamorphic:366: SELECT operation, message FROM [SHOW KV TRACE FOR SESSION]
        WHERE message     LIKE '%r44: sending batch%'
          AND message NOT LIKE '%PushTxn%'
          AND message NOT LIKE '%QueryTxn%'
        expected:
            dist sender send  r44: sending batch 6 CPut to (n1,s1):1
            dist sender send  r44: sending batch 6 CPut to (n1,s1):1
            dist sender send  r44: sending batch 6 CPut to (n1,s1):1
            dist sender send  r44: sending batch 6 CPut to (n1,s1):1
            dist sender send  r44: sending batch 1 EndTxn to (n1,s1):1
        but found (query options: "") :
            dist sender send  r44: sending batch 10 CPut to (n1,s1):1
            dist sender send  r44: sending batch 10 CPut to (n1,s1):1
            dist sender send  r44: sending batch 4 CPut, 1 EndTxn to (n1,s1):1
--- done: testdata/show_trace_nonmetamorphic with config local: 48 tests, 1 failures
--- done: testdata/window with config local: 22 tests, 0 failures
            --- FAIL: TestExecBuild/local/show_trace_nonmetamorphic/system_table_lookup (1.10s)
=== RUN   TestExecBuild/local/show_trace_nonmetamorphic
=== PAUSE TestExecBuild/local/show_trace_nonmetamorphic
=== CONT  TestExecBuild/local/show_trace_nonmetamorphic
        --- FAIL: TestExecBuild/local/show_trace_nonmetamorphic (3.61s)
=== RUN   TestExecBuild/local
    --- FAIL: TestExecBuild/local (0.00s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/opt/exec/execbuilder.TestExecBuild [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3753303&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3753303&tab=artifacts#/) on master @ [1a078671083946342ecf610a6fb899df2d1783d3](https://github.com/cockroachdb/cockroach/commits/1a078671083946342ecf610a6fb899df2d1783d3):


```
=== RUN   TestExecBuild
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestExecBuild445443747
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestExecBuild
    logic.go:3514: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestExecBuild445443747
--- FAIL: TestExecBuild (79.08s)
=== RUN   TestExecBuild/local/show_trace_nonmetamorphic/system_table_lookup
    logic.go:2441: let $rangeid = 44
--- done: testdata/window with config local: 22 tests, 0 failures
--- done: testdata/virtual with config local: 10 tests, 0 failures
=== CONT  TestExecBuild/local/show_trace_nonmetamorphic/system_table_lookup
    logic.go:2400: 
         
        testdata/show_trace_nonmetamorphic:366: SELECT operation, message FROM [SHOW KV TRACE FOR SESSION]
        WHERE message     LIKE '%r44: sending batch%'
          AND message NOT LIKE '%PushTxn%'
          AND message NOT LIKE '%QueryTxn%'
        expected:
            dist sender send  r44: sending batch 6 CPut to (n1,s1):1
            dist sender send  r44: sending batch 6 CPut to (n1,s1):1
            dist sender send  r44: sending batch 6 CPut to (n1,s1):1
            dist sender send  r44: sending batch 6 CPut to (n1,s1):1
            dist sender send  r44: sending batch 1 EndTxn to (n1,s1):1
        but found (query options: "") :
            dist sender send  r44: sending batch 24 CPut, 1 EndTxn to (n1,s1):1
--- done: testdata/show_trace_nonmetamorphic with config local: 48 tests, 1 failures
--- total progress: 319 statements/queries
            --- FAIL: TestExecBuild/local/show_trace_nonmetamorphic/system_table_lookup (1.58s)
=== RUN   TestExecBuild/local/show_trace_nonmetamorphic
=== PAUSE TestExecBuild/local/show_trace_nonmetamorphic
=== CONT  TestExecBuild/local/show_trace_nonmetamorphic
        --- FAIL: TestExecBuild/local/show_trace_nonmetamorphic (5.46s)
=== RUN   TestExecBuild/local
    --- FAIL: TestExecBuild/local (0.00s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/opt/exec/execbuilder.TestExecBuild [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3754851&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3754851&tab=artifacts#/) on master @ [0b9208b8e293a475d5a4c940b493c8414dfa95ea](https://github.com/cockroachdb/cockroach/commits/0b9208b8e293a475d5a4c940b493c8414dfa95ea):


```
=== RUN   TestExecBuild
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestExecBuild3407763170
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestExecBuild
    logic.go:3514: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestExecBuild3407763170
--- FAIL: TestExecBuild (78.20s)
=== RUN   TestExecBuild/local/show_trace_nonmetamorphic/system_table_lookup
    logic.go:2441: let $rangeid = 44
=== CONT  TestExecBuild/local/show_trace_nonmetamorphic/system_table_lookup
    logic.go:2400: 
         
        testdata/show_trace_nonmetamorphic:366: SELECT operation, message FROM [SHOW KV TRACE FOR SESSION]
        WHERE message     LIKE '%r44: sending batch%'
          AND message NOT LIKE '%PushTxn%'
          AND message NOT LIKE '%QueryTxn%'
        expected:
            dist sender send  r44: sending batch 6 CPut to (n1,s1):1
            dist sender send  r44: sending batch 6 CPut to (n1,s1):1
            dist sender send  r44: sending batch 6 CPut to (n1,s1):1
            dist sender send  r44: sending batch 6 CPut to (n1,s1):1
            dist sender send  r44: sending batch 1 EndTxn to (n1,s1):1
        but found (query options: "") :
            dist sender send  r44: sending batch 11 CPut to (n1,s1):1
            dist sender send  r44: sending batch 11 CPut to (n1,s1):1
            dist sender send  r44: sending batch 2 CPut, 1 EndTxn to (n1,s1):1
--- done: testdata/show_trace_nonmetamorphic with config local: 48 tests, 1 failures
--- progress: testdata/partial_index_nonmetamorphic: 38 statements/queries
--- done: testdata/inverted_join_multi_column with config local: 7 tests, 0 failures
            --- FAIL: TestExecBuild/local/show_trace_nonmetamorphic/system_table_lookup (3.15s)
=== RUN   TestExecBuild/local/show_trace_nonmetamorphic
=== PAUSE TestExecBuild/local/show_trace_nonmetamorphic
=== CONT  TestExecBuild/local/show_trace_nonmetamorphic
        --- FAIL: TestExecBuild/local/show_trace_nonmetamorphic (12.02s)
=== RUN   TestExecBuild/local
    --- FAIL: TestExecBuild/local (0.00s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/opt/exec/execbuilder.TestExecBuild [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3760128&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3760128&tab=artifacts#/) on master @ [28bb1ea049da5bfb6e15a7003cd7b678cbc4b67f](https://github.com/cockroachdb/cockroach/commits/28bb1ea049da5bfb6e15a7003cd7b678cbc4b67f):


```
=== RUN   TestExecBuild
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestExecBuild2951121496
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestExecBuild
    logic.go:3514: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestExecBuild2951121496
--- FAIL: TestExecBuild (75.76s)
=== RUN   TestExecBuild/local/show_trace_nonmetamorphic/system_table_lookup
=== CONT  TestExecBuild/local/show_trace_nonmetamorphic/system_table_lookup
    logic.go:2441: let $rangeid = 44
--- done: testdata/select_index with config local: 170 tests, 0 failures
--- progress: testdata/secondary_index_column_families_nonmetamorphic: 34 statements/queries
--- done: testdata/values with config local: 5 tests, 0 failures
=== CONT  TestExecBuild/local/show_trace_nonmetamorphic/system_table_lookup
    logic.go:2400: 
         
        testdata/show_trace_nonmetamorphic:366: SELECT operation, message FROM [SHOW KV TRACE FOR SESSION]
        WHERE message     LIKE '%r44: sending batch%'
          AND message NOT LIKE '%PushTxn%'
          AND message NOT LIKE '%QueryTxn%'
        expected:
            dist sender send  r44: sending batch 6 CPut to (n1,s1):1
            dist sender send  r44: sending batch 6 CPut to (n1,s1):1
            dist sender send  r44: sending batch 6 CPut to (n1,s1):1
            dist sender send  r44: sending batch 6 CPut to (n1,s1):1
            dist sender send  r44: sending batch 1 EndTxn to (n1,s1):1
        but found (query options: "") :
            dist sender send  r44: sending batch 24 CPut, 1 EndTxn to (n1,s1):1
--- done: testdata/show_trace_nonmetamorphic with config local: 48 tests, 1 failures
--- done: testdata/topk with config local: 7 tests, 0 failures
--- progress: testdata/partial_index_nonmetamorphic: 94 statements/queries
--- progress: testdata/upsert: 35 statements/queries
--- done: testdata/update with config local: 52 tests, 0 failures
            --- FAIL: TestExecBuild/local/show_trace_nonmetamorphic/system_table_lookup (4.07s)
=== RUN   TestExecBuild/local/show_trace_nonmetamorphic
=== PAUSE TestExecBuild/local/show_trace_nonmetamorphic
=== CONT  TestExecBuild/local/show_trace_nonmetamorphic
        --- FAIL: TestExecBuild/local/show_trace_nonmetamorphic (14.33s)
=== RUN   TestExecBuild/local
    --- FAIL: TestExecBuild/local (0.01s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/logictest/logic.go
+++ b/pkg/sql/logictest/logic.go
@@ -3482,8 +3482,10 @@ func RunLogicTestWithDefaultConfig(
 					if *printErrorSummary {
 						defer lt.printErrorSummary()
 					}
-					serverArgs.forceProductionBatchSizes = onlyNonMetamorphic
-					lt.setup(cfg, serverArgs, readClusterOptions(t, path))
+					// Each test needs a copy because of Parallel
+					serverArgsCopy := serverArgs
+					serverArgsCopy.forceProductionBatchSizes = onlyNonMetamorphic
+					lt.setup(cfg, serverArgsCopy, readClusterOptions(t, path))
 					lt.runFile(path, cfg)
 
 					progress.Lock()

